sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -467,4 +467,15 @@ object ResolveDefaultColumns extends QueryErrorsBase {
       v1Catalog.isPersistentFunction(ident.asFunctionIdentifier)
     }
   }
+
+
+  /**
+   * These define and cache existence default values for the struct fields for efficiency purposes.
```

Comment:
This does not appear to cache the results of each method, as this comment suggests. The previous code did this with a `lazy val`. Can you update the comment to mention that these results are *not* cached and that callers should take care to avoid calling them repeatedly in tight loops during query execution (e.g. per-row) and instead prefer to initialize them as infrequently as possible, preferably only once?